platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -186,15 +186,21 @@ private void uploadCoreConfiguration(String collection, CloudSolrClient client)
       Path configPath =
           Paths.get(configProxy.getDataDirectory().getAbsolutePath(), collection, "conf");
 
-      try {
-        client.uploadConfig(configPath, collection);
+      try (ZkClientClusterStateProvider zkStateProvider = newZkStateProvider(client)) {
+        zkStateProvider.uploadConfig(configPath, collection);
       } catch (IOException e) {
         throw new SolrFactoryException(
             "Failed to upload configurations for collection: " + collection, e);
       }
     }
   }
 
+  @VisibleForTesting
+  ZkClientClusterStateProvider newZkStateProvider(CloudSolrClient client) {
+    return new ZkClientClusterStateProvider(
+        client.getZkStateReader().getZkClient().getZkServerAddress());
+  }
+
   @SuppressWarnings({
     "deprecation" /* Pre-existing code until redesigned */,
```

Comment:
❓ Are those suppressions still required with the changes?

Reply:
Good catch.  Looks like these can be removed now.